### PR TITLE
Add Smoothing and Subsampling options for Tortuosity Computation

### PR DESCRIPTION
**Pull request by matthieuheitz**
Add UI to tweak the parameters
Bump TubeTK to bring smoothing and subsampling functions

### CODE REVIEW

**Comment by matthieuheitz** [comment]
Can someone review ?

File: Modules/Loadable/TortuosityModule/Logic/vtkSlicerTortuosityLogic.h [hunk 1]
```diff
@@ -88,7 +91,9 @@ class VTK_SLICER_TORTUOSITY_MODULE_LOGIC_EXPORT vtkSlicerTortuosityLogic
   bool RunSumOfAnglesMetric(vtkMRMLSpatialObjectsNode* node);
 
   // Run the metric specified by the flag on the given spatial object node.
-  bool RunMetrics(vtkMRMLSpatialObjectsNode* node, int flag);
+  bool RunMetrics(vtkMRMLSpatialObjectsNode* node, int flag, double smoothingScale = 0,
+                  tube::SmoothTubeFunctionEnum smoothingMethod = tube::SMOOTH_TUBE_USING_INDEX_GAUSSIAN,
```
Comment (aylward) [inline Modules/Loadable/TortuosityModule/Logic/vtkSlicerTortuosityLogic.h]: Always indent by 2 spaces, regardless of nested parenthesis, etc.

File: Modules/Loadable/TortuosityModule/Logic/vtkSlicerTortuosityLogic.cxx [hunk 2]
```diff
@@ -210,7 +213,8 @@ ::RunSumOfAnglesMetric(vtkMRMLSpatialObjectsNode* node)
 
 //------------------------------------------------------------------------------
 bool vtkSlicerTortuosityLogic
-::RunMetrics(vtkMRMLSpatialObjectsNode* node, int flag)
+::RunMetrics(vtkMRMLSpatialObjectsNode* node, int flag, double smoothingScale,
+             tube::SmoothTubeFunctionEnum smoothingMethod, int subsampling )
```
Comment (aylward) [inline Modules/Loadable/TortuosityModule/Logic/vtkSlicerTortuosityLogic.cxx]: Indent by 2

File: Modules/Loadable/TortuosityModule/Logic/vtkSlicerTortuosityLogic.cxx [hunk 3]
```diff
@@ -262,13 +266,21 @@ ::RunMetrics(vtkMRMLSpatialObjectsNode* node, int flag)
...
+
     // Update filter
     FilterType::Pointer filter = FilterType::New();
     filter->SetMeasureFlag(flag);
-    filter->SetInput(currTube);
+    filter->SetInput(subsampledTube);
     filter->Update();
 
-    if (filter->GetOutput()->GetId() != currTube->GetId())
+    if (filter->GetOutput()->GetId() != subsampledTube->GetId())
```
Comment (aylward) [inline Modules/Loadable/TortuosityModule/Logic/vtkSlicerTortuosityLogic.cxx]: Formatting of the if statement should be
if( filter->GetOutput().... )
- No space between if and right paren
- Space after right paren
- Space before left paren

File: Modules/Loadable/TortuosityModule/Widgets/qSlicerTortuosityModuleWidget.cxx [hunk 4]
```diff
@@ -136,8 +152,20 @@ ::setCurrentSpatialObjectsNode(vtkMRMLSpatialObjectsNode* node)
...
+  tube::SmoothTubeFunctionEnum smoothMethod =
+      (tube::SmoothTubeFunctionEnum)(d->SmoothingMethodComboBox->
+      itemData(d->SmoothingMethodComboBox->currentIndex()).toInt());
+
+  //Subsampling Factor
+  int subsampling = d->SubsamplingSliderWidget->value();
+
   d->RunPushButton->setEnabled(false);
-  if (!d->logic()->RunMetrics(d->currentSpatialObject, flag))
+  if (!d->logic()->RunMetrics(d->currentSpatialObject, flag, h, smoothMethod, subsampling))
```
Comment (aylward) [inline Modules/Loadable/TortuosityModule/Widgets/qSlicerTortuosityModuleWidget.cxx]: if( xxxxx )

**Comment by aylward** [comment]
Looks good, except a few small style issues.  Comments are given inline.

On Wed, Apr 8, 2015 at 10:12 AM, Matthieu Heitz <notifications@github.com>
wrote:

> Can someone review ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KitwareMedical/VesselView/pull/96#issuecomment-90928984>
> .
>



-- 
====================================================
Stephen R. Aylward, Ph.D.
Senior Director of Operations, North Carolina, Kitware, Inc.
http://www.kitware.com and http://www.aylward.org
(919) 969-6990 x300
====================================================

File: Modules/Loadable/TortuosityModule/Logic/vtkSlicerTortuosityLogic.cxx (hunk 3 above)
Comment (jcfr) [inline Modules/Loadable/TortuosityModule/Logic/vtkSlicerTortuosityLogic.cxx]: Should VTK derived class following VTK style ? 
See https://docs.google.com/a/kitware.com/document/d/1nzinw-dR5JQRNi_gb8qwLL5PnkGMK2FETlQGLr10tZw/edit

Comment (aylward) [inline Modules/Loadable/TortuosityModule/Logic/vtkSlicerTortuosityLogic.cxx]: Good point.

Go with that for vtk-derived classes.

On Wed, Apr 8, 2015 at 11:13 AM, Jean-Christophe Fillion-Robin <
notifications@github.com> wrote:

> In Modules/Loadable/TortuosityModule/Logic/vtkSlicerTortuosityLogic.cxx
> <https://github.com/KitwareMedical/VesselView/pull/96#discussion_r27979205>
> :
>
> >      filter->Update();
> >
> > -    if (filter->GetOutput()->GetId() != currTube->GetId())
> > +    if (filter->GetOutput()->GetId() != subsampledTube->GetId())
>
> Should VTK derived class following VTK style ?
> See
> https://docs.google.com/a/kitware.com/document/d/1nzinw-dR5JQRNi_gb8qwLL5PnkGMK2FETlQGLr10tZw/edit
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KitwareMedical/VesselView/pull/96/files#r27979205>.
>



-- 
====================================================
Stephen R. Aylward, Ph.D.
Senior Director of Operations, North Carolina, Kitware, Inc.
http://www.kitware.com and http://www.aylward.org
(919) 969-6990 x300
====================================================

**Pull request merged**